packages/ipfs-daemon/src/ipfs-daemon.ts

```diff
@@ -91,19 +89,20 @@ export class IpfsDaemon {
         const ipfs = await IPFS.create({
             start: false,
             repo,
+            // TODO: @rvagg dag-jose?
             // eslint-disable-next-line @typescript-eslint/ban-ts-comment
             // @ts-ignore
             ipld: {formats: [format]},
             libp2p: {
-                config: {
-                    dht: {
-                        enabled: false,
-                        clientMode: !configuration.ipfsDhtServerMode,
-                        randomWalk: false,
-                    },
-                    pubsub: {
-                        enabled: configuration.ipfsEnablePubsub,
-                    },
+                dht: {
+                    enabled: false,
+                    clientMode: !configuration.ipfsDhtServerMode,
+                    randomWalk: false,
+                },
+                pubsub: {
+                    // TODO: @rvagg is this gone?
+                    // @ts-ignore
+                    enabled: configuration.ipfsEnablePubsub,
```

Comment:
Were we doing this wrong or did it change?

Reply:
@vasco-santos can you answer this one? I can't get to the bottom of the change here. `dht` and `pubsub` have been lifted out of `config` to the top level in the `libp2p` config options. Was that always wrong here, or a recent change, or a problem with types?

Reply:
This is still inside the config, so this should not really be changed. Were there type errors configuring inside the config property?

Reply:
The in progress branch of `js-ipfs` with the types improvements also has it inside the config, so the types should be correct upstream: https://github.com/ipfs/js-ipfs/blob/chore/update-buffer-deps/packages/ipfs-core/src/runtime/libp2p-nodejs.js#L42

Reply:
See ipfs-core/src/types.d.ts and how it pulls in `Libp2pConfig` rather than `Libp2pOptions`. I _think_ this is the source of the problem but you'll have to confirm. https://github.com/ipfs/js-ipfs/issues/3640